frontend/src/pages/CompareV2.tsx

```diff
@@ -149,15 +196,110 @@ function getRunArtifacts(runs: ApiRunDetail[], mlmdPackages: MlmdPackage[]): Run
+      <p className={classes(css.visualizationPlaceholderText)}>
+        The selected {metricsTabText} will be displayed here.
+      </p>
+    </div>
+  );
+}
+
+interface MetricsDropdownProps {
+  filteredRunArtifacts: RunArtifact[];
+  metricsTabText: string;
+}
+
+const logDisplayNameWarning = (type: string, id: number | string) =>
+  logger.warn(`Failed to fetch the display name of the ${type} with the following ID: ${id}`);
+
+const getExecutionName = (execution: Execution) =>
+  execution
+    .getCustomPropertiesMap()
+    .get('display_name')
+    ?.getStringValue();
+
+function getDropdownItems(filteredRunArtifacts: RunArtifact[]) {
+  const dropdownItems: DropdownItem[] = [];
+  for (const runArtifact of filteredRunArtifacts) {
+    const runName = runArtifact.run.run?.name;
+    if (runName) {
+      // Combine execution names and artifact names into the same dropdown sub item.
+      const subItems: DropdownSubItem[] = [];
+      for (const executionArtifact of runArtifact.executionArtifacts) {
+        const executionName = getExecutionName(executionArtifact.execution);
+        if (executionName) {
+          // Group each artifact name with its parent execution name.
+          const executionLinkedArtifacts: DropdownSubItem[] = [];
+          for (const linkedArtifact of executionArtifact.linkedArtifacts) {
+            const artifactName = getArtifactName(linkedArtifact);
+            if (artifactName) {
+              executionLinkedArtifacts.push({
+                name: executionName,
+                secondaryName: artifactName,
+              } as DropdownSubItem);
+            } else {
+              logDisplayNameWarning('artifact', linkedArtifact.artifact.getId());
+            }
+          }
+          subItems.push(...executionLinkedArtifacts);
+        } else {
+          logDisplayNameWarning('execution', executionArtifact.execution.getId());
+        }
+      }
+
+      if (subItems.length > 0) {
+        dropdownItems.push({
+          name: runName,
+          subItems,
+        } as DropdownItem);
+      }
+    } else {
+      logDisplayNameWarning('run', runArtifact.run.run!.id!);
+    }
+  }
+
+  return dropdownItems;
+}
+
+function MetricsDropdown(props: MetricsDropdownProps) {
+  const { filteredRunArtifacts, metricsTabText } = props;
+  const [selectedItem, setSelectedItem] = useState<SelectedItem>({ itemName: '', subItemName: '' });
+
+  const dropdownItems: DropdownItem[] = getDropdownItems(filteredRunArtifacts);
+
+  if (dropdownItems.length === 0) {
+    return <p>There are no {metricsTabText} artifacts available on the selected runs.</p>;
+  }
+
+  return (
+    <>
+      <TwoLevelDropdown
+        title={`Choose a first ${metricsTabText} artifact`}
+        items={dropdownItems}
+        selectedItem={selectedItem}
+        setSelectedItem={setSelectedItem}
+      />
+      <VisualizationPlaceholder metricsTabText={metricsTabText} />
+    </>
+  );
+}
+
 function CompareV2(props: PageProps) {
```

Comment:
This is a good question, I only named it `CompareV2` because the original page was named `Compare` so I kept that format. If I change this I'll change the other file names as well (`Compare`, `CompareV1`, `CompareV2`, and associated test files). @zijianjoy do you have any preference?

Reply:
It is a good point but Chris is only following the current naming pattern in this PR. We can make the naming pattern change in a separate PR.